apps/www/registry/default/ui/multi-select.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from "react"
+import { cn } from "lib/utils"
+import { Check, ChevronsUpDown, X } from "lucide-react"
+
+import { Badge } from "@/registry/default/ui/badge"
+import { Button } from "@/registry/default/ui/button"
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+} from "@/registry/default/ui/command"
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/registry/default/ui/popover"
+
+export type OptionType = Record<"value" | "label", string>
+
+interface MultiSelectProps {
+  options: Record<"value" | "label", string>[]
+  selected: Record<"value" | "label", string>[]
+  onChange: React.Dispatch<
+    React.SetStateAction<Record<"value" | "label", string>[]>
+  >
+  className?: string
+  placeholder?: string
+}
+
+const MultiSelect = React.forwardRef<HTMLButtonElement, MultiSelectProps>(
+  ({ options, selected, onChange, className, ...props }, ref) => {
+    const [open, setOpen] = React.useState(false)
+    const [query, setQuery] = React.useState<string>("")
+
+    const handleUnselect = (item: Record<"value" | "label", string>) => {
+      onChange(selected.filter((i) => i.value !== item.value))
+    }
+
+    // on delete key press, remove last selected item
+    React.useEffect(() => {
+      const handleKeyDown = (e: KeyboardEvent) => {
+        if (e.key === "Backspace" && query === "" && selected.length > 0) {
+          onChange(selected.filter((_, index) => index !== selected.length - 1))
+        }
+
+        // close on escape
+        if (e.key === "Escape") {
+          setOpen(false)
+        }
+      }
+
+      document.addEventListener("keydown", handleKeyDown)
+
+      return () => {
+        document.removeEventListener("keydown", handleKeyDown)
+      }
+    }, [onChange, query, selected])
+
+    return (
+      <Popover open={open} onOpenChange={setOpen}>
+        <PopoverTrigger asChild className={className}>
+          <Button
+            ref={ref}
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className={`group w-full justify-between ${
+              selected.length > 1 ? "h-full" : "h-10"
+            }`}
+            onClick={() => setOpen(!open)}
+          >
+            <div className="flex flex-wrap items-center gap-1">
+              {selected.map((item) => (
+                <Badge
+                  variant="outline"
+                  key={item.value}
+                  className="flex items-center gap-1 group-hover:bg-background"
+                >
+                  {item.label}
+                  {open && (
+                    <Button
+                      asChild
+                      variant="outline"
+                      size="icon"
+                      className={`border-none duration-300 ${
+                        open ? "opacity-100 ease-in" : "opacity-0 ease-out"
```

Comment:
Unnecessary conditional, `open` value is always truthy
because button only visible when open is true
```suggestion
                       opacity-100 ease-in
```